org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/jvmmodel/JvmModelAssociator.java

```diff
@@ -467,6 +467,9 @@ public void removeAssociation(EObject sourceElement, EObject jvmElement) {
 		if (sources != null && sources.remove(sourceElement)) {
 			Set<EObject> targets = sourceToTargetMap(resource).get(sourceElement);
 			targets.remove(jvmElement);
+			if (sources.isEmpty()) {
```

Comment:
have no idea how to test this in
org.eclipse.xtext.xbase.tests.jvmmodel.JvmModelAssociatorTest.testRemoveAssociation()
as it always will return empty set.

Reply:
am not sure about removeAssocian vs removeAllAssociation

Reply:
testcase in ResourceStorageTest
does not call remove*Assoc at all.